diff --git a/esm/CustomMenuPlugin.js b/esm/CustomMenuPlugin.js
index d630c6893dd77f148ca72cf2b137364bbce3830c..eb0767f52e9654c915d606fdd330b7c942daa8d3 100644
--- a/esm/CustomMenuPlugin.js
+++ b/esm/CustomMenuPlugin.js
@@ -622,7 +622,7 @@ export class CustomMenuView {
                         const mousedownEvent = new MouseEvent('mousedown', {
                             bubbles: true,
                             cancelable: true,
-                            view: dntShim.dntGlobalThis,
+                            view: this.editorView.dom.ownerDocument.defaultView,
                         });
                         dom.dispatchEvent(mousedownEvent);
                     }
@@ -759,7 +759,7 @@ export class CustomMenuView {
                             const mousedownEvent = new MouseEvent('mousedown', {
                                 bubbles: true,
                                 cancelable: true,
-                                view: dntShim.dntGlobalThis,
+                                view: this.editorView.dom.ownerDocument.defaultView,
                             });
                             dom.dispatchEvent(mousedownEvent);
                         }
@@ -771,7 +771,7 @@ export class CustomMenuView {
                             const mousedownEvent = new MouseEvent('mousedown', {
                                 bubbles: true,
                                 cancelable: true,
-                                view: dntShim.dntGlobalThis,
+                                view: this.editorView.dom.ownerDocument.defaultView,
                             });
                             dom.dispatchEvent(mousedownEvent);
                         }
diff --git a/src/CustomMenuPlugin.ts b/src/CustomMenuPlugin.ts
index bddaf1c8de850ef407ca40d5fadcc1ac71a8dca7..3ae2e52d0b27d98fb0f3d87decf24f1de3f28a86 100644
--- a/src/CustomMenuPlugin.ts
+++ b/src/CustomMenuPlugin.ts
@@ -742,7 +742,7 @@ export class CustomMenuView {
             const mousedownEvent = new MouseEvent('mousedown', {
               bubbles: true,
               cancelable: true,
-              view: dntShim.dntGlobalThis,
+              view: this.editorView.dom.ownerDocument.defaultView,
             });
             dom.dispatchEvent(mousedownEvent);
           }
@@ -905,7 +905,7 @@ export class CustomMenuView {
               const mousedownEvent = new MouseEvent('mousedown', {
                 bubbles: true,
                 cancelable: true,
-                view: dntShim.dntGlobalThis,
+                view: this.editorView.dom.ownerDocument.defaultView,
               });
               dom.dispatchEvent(mousedownEvent);
             }
@@ -918,7 +918,7 @@ export class CustomMenuView {
               const mousedownEvent = new MouseEvent('mousedown', {
                 bubbles: true,
                 cancelable: true,
-                view: dntShim.dntGlobalThis,
+                view: this.editorView.dom.ownerDocument.defaultView,
               });
               dom.dispatchEvent(mousedownEvent);
             }
